.github/workflows/milestone.yml

```diff
@@ -14,13 +13,26 @@ jobs:
         uses: actions/checkout@v2
       - name: Get milestone from pom.xml
         run: |
-          ./mvnw -v
+          .github/bin/retry ./mvnw -v
           echo "MILESTONE_NUMBER=$(./mvnw -q -Dexec.executable=echo -Dexec.args='${project.version}' -N exec:exec | cut -d- -f1)" >> $GITHUB_ENV
       - name: Set milestone to PR
-        uses: actions/github-script@v3
+        uses: actions/github-script@v4
         with:
           github-token: ${{secrets.GITHUB_TOKEN}}
           script: |
+            // Get pull request number from commit sha
+            // 'listPullRequestsAssociatedWithCommit()' lists the merged pull request
+            // https://docs.github.com/en/rest/reference/repos#list-pull-requests-associated-with-a-commit
+            const pr_response = await github.repos.listPullRequestsAssociatedWithCommit({ owner: context.repo.owner, repo:context.repo.repo, commit_sha: context.sha })
+            if (pr_response.data.length === 0) {
+              console.log('Pull request not found')
+              return
+            }
+            if (pr_response.data.length > 1) {
+              console.log(pr_response.data)
+              throw 'Expect 1 pull request but found: ' + pr_response.data.length
```

Comment:
Can we list the pull requests that were found? It might help in debugging if we find some issues.

Reply:
The above line `console.log(pr_response.data)` at L32 contains such information :) 
It's json object, so I didn't include it in `throw` message to avoid dirty message.